cmd2/cmd2.py

```diff
@@ -1581,6 +1578,24 @@ def complete(self, text: str, state: int) -> Optional[str]:
         except IndexError:
             return None
 
+    def complete(self, text: str, state: int) -> Optional[str]:
+        """Override of cmd2's complete method which returns the next possible completion for 'text'
+
+        This method gets called directly by readline. Since readline suppresses any exception raised
+        in completer functions, they can be difficult to debug. Therefore this function wraps the
+        actual tab completion logic and prints to stderr any exception that occurs before returning
+        control to readline.
+
+        :param text: the current word that user is typing
+        :param state: non-negative integer
+        """
+        # noinspection PyBroadException
+        try:
+            return self._complete_worker(text, state)
+        except Exception as e:
+            self.perror(e)
```

Comment:
This should be a big help to people for developers that are trying to debug tab-completion issues.